src/cpu.c

```diff
@@ -128,6 +141,53 @@ int clockCpu(struct CPU *cpu, struct BUS *bus) {
       case STY:
         writeBus(address, cpu->index_y, bus);
         break;
+      case NOP:
+        printf("let's slide!");
+        break;
+      case LDA:
+        cpu->accumulator = readBus(address, bus);
+        //todo: this should be removed, copy from CPY and create a method
+        setZeroAndNegativeFlags(cpu, cpu->accumulator);
+        break;
+      case TXA: 
+        cpu->index_x = cpu->accumulator;
+        //todo: this should be removed, copy from CPY and create a method
+        setZeroAndNegativeFlags(cpu, cpu->index_x);
+        break;
+      case INX:
+        cpu->index_x++;
+        //todo: this should be removed, copy from CPY and create a method
+        setZeroAndNegativeFlags(cpu, cpu->index_x);
+        break;
+      case INY:
+        cpu->index_y++;
+        //todo: this should be removed, copy from CPY and create a method
+        setZeroAndNegativeFlags(cpu, cpu->index_y);
+ 
+        break;
+      case CPY: 
+        compareResult = (signed char)cpu->index_y - readBus(address, bus);
+
+        cpu->status.carry = compareResult >= 0;
+        cpu->status.zero = compareResult == 0;
+        cpu->status.negative = compareResult < 0;
+        break;
+      case PHA:
+        pushStack(cpu, bus, cpu->accumulator);
+        break;
+      case LDY:
+        cpu->index_y = bus->memory[address];
+        //todo: this should be removed, copy from CPY and create a method
+        setZeroAndNegativeFlags(cpu, cpu->index_y);
+        break;
+      case LDX:
+        cpu->index_x = bus->memory[address];
+        //todo: this should be removed, copy from CPY and create a method
+        setZeroAndNegativeFlags(cpu, cpu->index_x);
+        break;
+      case BNE:
+        if (!cpu->status.zero) cpu->pc = address; // this address was already understood on the method `getAddressByOpcode`
```

Comment:
Maybe rename this var and remove this comment?

Reply:
Yes, i'll remove it but which var?

Reply:
address

Reply:
Probably the `relativeAddressMode` method is incorrect... 
If that's the case, we don't need to rename the `address` var.
Makes any sense?

Reply:
Ah ok

Reply:
Let me think on it for a moment... I'm looking at this method now and it's correct, I mean, when relative, the PC needs to be added with the operand address.
So, when we hit this `if`statement, we really has an address to simply pass to PC and change the program flow.
What you think?